src/containers/Contacts/ContactForm.tsx

```diff
@@ -0,0 +1,157 @@
+import { addContact } from '@actions'
+import { InputField } from '@components/InputField'
+import { RootState } from '@store'
+import { Contact } from '@types'
+import { Keypair } from 'js-kinesis-sdk'
+import * as React from 'react'
+import { connect } from 'react-redux'
+
+interface State extends Contact {
+  errors: { [key in keyof Contact]?: string }
+  isValid: boolean
+}
+
+const mapStateToProps = (state: RootState) => ({
+  currentContacts: state.contacts.contactList,
+})
+
+const mapDispatchToProps = {
+  addContact,
+}
+
+type StateFulProps = { hideForm: () => any } & typeof mapDispatchToProps &
+  ReturnType<typeof mapStateToProps>
+class ContractFormStateful extends React.Component<StateFulProps, State> {
+  state: State = {
+    name: '',
+    address: '',
+    errors: {},
+    isValid: false,
+  }
+
+  render() {
+    return (
+      <ContactFormPresentation
+        {...this.state}
+        handleChange={this.handleChange}
+        cancel={this.props.hideForm}
+        addContact={this.handleAdd}
+      />
+    )
+  }
+
+  private handleAdd = () => {
+    const possibleErrors = this.validateForm()
+    this.setState({ errors: possibleErrors })
+
+    const isValid = Object.values(possibleErrors).every(error => error === '')
+    if (!isValid) {
+      return
+    }
+
+    const { name, address } = this.state
+    this.props.addContact({ name, address })
+    this.props.hideForm()
+  }
+
+  private handleChange = (key: string, value: string) => {
+    this.setState(state => ({ ...state, [key]: value }))
+  }
+
+  private validateForm = () => {
+    const errors = {
+      address:
+        (this.validateAddress() ? '' : 'Invalid address') ||
+        (this.props.currentContacts.every(({ address }) => this.state.address !== address)
```

Comment:
this could also be inverted to use `some` as below, instead of going through all of them with the `every` call?

Reply:
The actual timing is the same, as they both will either fail early or have to evaluate each.